elastalert/elastalert.py

```diff
@@ -1067,7 +1067,13 @@ def load_rule_changes(self):
             if rule_file not in new_rule_hashes:
                 # Rule file was deleted
                 elastalert_logger.info('Rule file %s not found, stopping rule execution' % (rule_file))
-                self.rules = [rule for rule in self.rules if rule['rule_file'] != rule_file]
+                for rule in self.rules:
+                    if rule['rule_file'] == rule_file:
+                        break
```

Comment:
what if there is more than one disabled rule? won't this only disable one of them?

Reply:
See line 1066. This is in a loop over all rule hashes